src/helper/proxy/index.ts

```diff
@@ -45,6 +68,7 @@ const buildRequestInitFromRequest = (
     method: request.method,
     body: request.body,
     duplex: request.body ? 'half' : undefined,
+    ...requestInit,
     headers,
```

Comment:
Previously the `headers` from `buildRequestInitFromRequest(raw)` were being overwritten by `...requestInit` on line 97

https://github.com/honojs/hono/blob/e7fbc3e2d60a657f5a330daa59b068073f061d5e/src/helper/proxy/index.ts#L95-L98